index.html

```diff
@@ -418,6 +419,145 @@ <h2>Transferable MediaStreamTrack</h2>
       <p>The underlying source is supposed to be kept alive between the transfer and transfer-receiving steps, or as long as the data holder is alive.
       In a sense, between these steps, the data holder is attached to the underlying source as if it was a track.</p>
     </div>
+    <h3>MediaStreamTrack frame counters</h3>
+    <div>
+      <p>
+        Frame counters allow you to tell what the capture frame rate is, which
+        may be different from what is expected from settings - for instance, the
+        camera may be slowing down frame production because it's dark or the
+        system may be CPU starved and therefore dropping frames.
+      </p>
+    </div>
+    <div>
+      <pre class="idl"
+>partial interface MediaStreamTrack {
+  Promise&lt;MediaStreamTrackStats&gt; getStats();
+};
+</pre>
+    </div>
+    <div>
+      <p>At creation of a {{MediaStreamTrack}} object, <var>track</var>, run the
+      following steps:</p>
+      <ol>
+        <li>
+          <p>Initialize <var>track</var>.`[[FramesCaptured]]` to 0.</p>
+        </li>
+      </ol>
+    </div>
+    <div>
+      <p>Monitor frames that arrive to <var>track</var> from its source by
+      running the following in-parallel steps for every <var>frame</var>:</p>
+      <ol>
+        <li>
+          <p>If <var>frame</var> is to be discarded by <var>track</var> in
+          order to achive the frame rate decimation of <var>track</var>'s
+          settings, abort these steps.</p>
+        </li>
+        <li>
+          <p>Atomically increment <var>track</var>.`[[FramesCaptured]]` by
+          1.</p>
+        </li>
+      </ol>
+    </div>
+    <section>
+      <h4>Methods</h4>
+      <dl data-link-for="MediaStreamTrack" data-dfn-for="MediaStreamTrack"
+      class="methods">
+        <dt>
+          <dfn>getStats</dfn>
+        </dt>
+        <dd>
+          <p>
+            When this method is called, the user agenst MUST run the following
+            steps:
+          </p>
+          <ol>
+            <li>
+              <p>
+                Let <var>track</var> be the {{MediaStreamTrack}} that this
+                method was called on.
+              </p>
+            </li>
+            <li>
+              <p>Let <var>p</var> be a new promise.</p>
+            </li>
+            <li>
+              <p>
+                In parallel, begin the [= in-parallel steps to get track
+                stats =], given <var>track</var> and <var>p</var>.
+              </p>
+            </li>
+            <li>
+              <p>Return <var>p</var>.</p>
+            </li>
+          </ol>
+          <p>
+            The <dfn>in-parallel steps to get track stats</dfn> given
+            <var>track</var> and <var>p</var> are as follows:
+          </p>
+          <ol>
+            <li>
+              <p>
+                Let <var>timestamp</var> be the current time relative to the
+                UNIX epoch (Jan 1, 1970, UTC).
```

Comment:
Is there any precedent for an API returning a timestamp that is basically now + the time it takes to queue a task on main thread? Seems of low value to me. What is the resolution of this timer? The only time it would differ substantially is if the main thread is under heavy load that tasks have already been queued ahead of of this task. 

To mitigate this, since the read is "atomic", have we considered reading it in the queued task? This would remove any need for a timestamp IMHO.

Reply:
RTCPeerConnection.getStats() is an example of using timestamp for the same reason, but if we can make an atomic read from the main thread then we can probably get rid of timestamp, but then the UA would no longer be allowed to implement this using a single threaded frame counter off the main thread, which I think would be nice to be allowed to do

Reply:
I changed the PR to be requirements-based and not have internal slots or "atomic" language, so now the timestamp is needed to avoid forcing how to implement this